### PR TITLE
[AUTO][OLM][OCP-24829] Report `Upgradeable` in OLM ClusterOperators status

### PR DESCRIPTION
**Pull request by bandrade**
PR Created to automate test case: @ecordell @njhale @kevinrizza. please have a review. Thanks! cc: @cuipinghuo @scolange @chengzhang1016 @emmajiafan @jianzhangbjz @tbuskey

Success execution evidence:

```
 ./openshift-tests run all --dry-run | grep "Report Upgradeable" | ./openshift-tests run -f -

I1226 04:35:16.456207   24579 test_context.go:419] Tolerating taints "node-role.kubernetes.io/master" when considering if nodes are ready
I1226 04:35:17.864259   24582 test_context.go:419] Tolerating taints "node-role.kubernetes.io/master" when considering if nodes are ready
started: (0/1/1) "[Feature:Platform] OLM should Report Upgradeable in OLM ClusterOperators status [Suite:openshift/conformance/parallel]"

I1226 04:35:35.374320   25071 test_context.go:419] Tolerating taints "node-role.kubernetes.io/master" when considering if nodes are ready
Dec 26 04:35:35.439: INFO: Waiting up to 30m0s for all (but 100) nodes to be schedulable
Dec 26 04:35:35.986: INFO: Waiting up to 10m0s for all pods (need at least 0) in namespace 'kube-system' to be running and ready
Dec 26 04:35:36.570: INFO: 0 / 0 pods in namespace 'kube-system' are running and ready (0 seconds elapsed)
Dec 26 04:35:36.570: INFO: expected 0 pod replicas in namespace 'kube-system', 0 are Running and Ready.
Dec 26 04:35:36.570: INFO: Waiting up to 5m0s for all daemonsets in namespace 'kube-system' to start
Dec 26 04:35:36.769: INFO: e2e test version: v0.0.0-master+$Format:%h$
Dec 26 04:35:36.920: INFO: kube-apiserver version: v1.17.0
Dec 26 04:35:37.078: INFO: Cluster IP family: ipv4
[BeforeEach] [Top Level]
  /home/bandrade/go/src/github.com/openshift/origin/test/extended/util/test.go:60
[BeforeEach] [Feature:Platform] OLM should
  /home/bandrade/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:154
STEP: Creating a kubernetes client
[It] Report Upgradeable in OLM ClusterOperators status [Suite:openshift/conformance/parallel]
  /home/bandrade/go/src/github.com/openshift/origin/test/extended/operators/olm.go:134
Dec 26 04:35:37.109: INFO: Running 'oc --config=/home/bandrade/QE/ocp4-aws-ipi-vagrant-4.4/cluster/auth/kubeconfig get co operator-lifecycle-manager -o=jsonpath={range .status.conditions[*]}{.type}'
Dec 26 04:35:38.363: INFO: Running 'oc --config=/home/bandrade/QE/ocp4-aws-ipi-vagrant-4.4/cluster/auth/kubeconfig get co operator-lifecycle-manager-catalog -o=jsonpath={range .status.conditions[*]}{.type}'
Dec 26 04:35:39.333: INFO: Running 'oc --config=/home/bandrade/QE/ocp4-aws-ipi-vagrant-4.4/cluster/auth/kubeconfig get co operator-lifecycle-manager-packageserver -o=jsonpath={range .status.conditions[*]}{.type}'
[AfterEach] [Feature:Platform] OLM should
  /home/bandrade/go/src/github.com/openshift/origin/test/extended/util/client.go:120
[AfterEach] [Feature:Platform] OLM should
  /home/bandrade/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:155
Dec 26 04:35:40.152: INFO: Waiting up to 7m0s for all (but 100) nodes to be ready
Dec 26 04:35:40.560: INFO: Running AfterSuite actions on all nodes
Dec 26 04:35:40.560: INFO: Running AfterSuite actions on node 1

passed: (22.6s) 2019-12-26T07:35:40 "[Feature:Platform] OLM should Report Upgradeable in OLM ClusterOperators status [Suite:openshift/conformance/parallel]"

1 pass, 0 skip (22.6s)

```

### CODE REVIEW

File: test/extended/operators/olm.go [hunk 1]
```diff
@@ -105,4 +105,18 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 		}
 	})
 
+	// OCP-24829 - Report `Upgradeable` in OLM ClusterOperators status
+	// author: bandrade@redhat.com
+	g.It("Report Upgradeable in OLM ClusterOperators status", func() {
+		olmCOs := []string{"operator-lifecycle-manager", "operator-lifecycle-manager-catalog", "operator-lifecycle-manager-packageserver"}
+		for _, co := range olmCOs {
+			msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("co", co, "-o=jsonpath={range .status.conditions[*]}{.type}").Output()
```
Comment (njhale) [inline test/extended/operators/olm.go]: Is the objective of this test to check whether OLM writes _any_ upgradeable status, or that it _is_ upgradeable? In the later case, you would also need to ensure the condition's `status` field is set `True`.

Comment (bandrade) [reply test/extended/operators/olm.go]: Fixed as requested

**Comment by njhale** [comment]
This looks good. I think my [question from the last review still holds](https://github.com/openshift/origin/pull/24352#pullrequestreview-337674256), but I don't think it should keep this PR from merging in the meantime.

/lgtm

**Comment by bandrade** [comment]
@njhale Thanks, I did a quick search but I didn't find a generic test for this matter.

**Comment by bandrade** [comment]
@shawn-hurley @bparees Could you help review it and give approve? Thanks!

**Comment by shawn-hurley** [comment]
```
oc get co operator-lifecycle-manager -o=jsonpath={range .status.conditions[*]}{.type}{' '}{.status}
zsh: no matches found: .status.conditions[*]}{.type}{ }{.status}
```

I am unable to get this to work locally, am I missing something?

**Comment by shawn-hurley** [comment]
Note to self, I got it to work once I put the quotes around it. 

/approve

**Comment by bandrade** [comment]
@shawn-hurley you may want to use  `oc get co operator-lifecycle-manager -o jsonpath="{range .status.conditions[*]}{.type}{' '}{.status}"` . On framework it has some diferences due to escape chars.

**Pull request merged**